### ISSUE TITLE
FormatException when typing "0" (zero) in incomplete statement

### ISSUE THREAD

**Issue by KevinH-MS** [dotnet/roslyn#6079]
![capture](https://cloud.githubusercontent.com/assets/6464209/10549058/89acd082-73f3-11e5-88ff-6bad091a7591.PNG)

Dump at:  [internal file share]\6079\FormatException.dmp
```
System.FormatException
>	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.ErrorReporting.FatalError.Report(System.Exception exception, System.Action<System.Exception> handler) + 0x38 bytes	
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.DocumentState.IncrementallyParseTreeAsync() + 0x5fb bytes	
 	mscorlib.dll!System.Number.StringToNumber(string str, System.Globalization.NumberStyles options, ref System.Number.NumberBuffer number, System.Globalization.NumberFormatInfo info, bool parseDecimal) + 0xed bytes	
 	mscorlib.dll!System.Number.ParseInt32(string s, System.Globalization.NumberStyles style, System.Globalization.NumberFormatInfo info) + 0x79 bytes	
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.RealParser.DecimalFloatingPointString.FromSource(string source) + 0x1a6 bytes	
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.RealParser.TryParseDouble(string s, out double d) + 0x21 bytes	
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Lexer.GetValueDouble(string text) + 0x1e bytes	
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Lexer.ScanNumericLiteral(ref Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Lexer.TokenInfo info) + 0x522 bytes	
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Lexer.ScanSyntaxToken(ref Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Lexer.TokenInfo info) + 0x5fe bytes	
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Lexer.LexSyntaxToken() + 0x69 bytes	
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Lexer.Lex(Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LexerMode mode) + 0x2f bytes	
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Blender.Reader.LexNewToken(Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LexerMode mode) + 0x5b bytes	
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Blender.Reader.ReadNewToken(Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LexerMode mode) + 0x15 bytes	
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Blender.Reader.ReadNodeOrToken(Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LexerMode mode, bool asToken) + 0x6d bytes	
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Blender.ReadNodeOrToken(Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LexerMode mode, bool asToken) + 0x8a bytes	
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.SyntaxParser.AddNewToken() + 0x73 bytes	
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.SyntaxParser.FetchCurrentToken() + 0x16 bytes	
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.SyntaxParser.CurrentToken.get() + 0x13 bytes
```

**Comment by KevinH-MS** [comment]
@gafter is this related to your recent changes?

**Comment by gafter** [comment]
Sure looks like it.

**Comment by gafter** [comment]
Fixed in the stabilization branch by #6084.
